vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -2532,9 +2532,25 @@ public void deselect(T item) {
      * @see GridSelectionModel
      */
     public void deselectAll() {
+        setDeselectAllowed(this,true);
         getSelectionModel().deselectAll();
     }
 
+    /**
+     * This method is to set DeselectAllowed
+     *
+     * @param grid
+     *           the current grid
+     * @param deselectAllowed
+     *           value to set for deselectAllowed
+     */
+    private void setDeselectAllowed(Grid<?> grid, boolean deselectAllowed) {
```

Comment:
1. Parameter grid is redundant, it will only be used with `this`.
1. The javadoc has no information that can't be deduced by reading the method declaration and is unnecessary for a private method.
1. Doesn't anything need to be done in multi selection?